src/plugins/socketmon/socketmon.cpp

```diff
@@ -1096,21 +933,42 @@ socketmon::socketmon(drakvuf_t drakvuf, const socketmon_config* c, output_format
         throw -1;
     }
 
-    event_response_t(*tcpe_cb)( drakvuf_t drakvuf, drakvuf_trap_info_t* info ) =
-        ((pm == VMI_PM_IA32E) ?
-         ((winver == VMI_OS_WINDOWS_10) ?
-          tcpe_win10_x64_cb :
-          tcpe_x64_cb) :
-         tcpe_x86_cb);
-
-    assert(sizeof(tcpip_traps) / sizeof(tcpip_traps[0]) > 6);
-    register_tcpip_trap(drakvuf, tcpip_profile_json, "TcpTcbDelay", &this->tcpip_traps[0], tcpe_cb);
-    register_tcpip_trap(drakvuf, tcpip_profile_json, "TcpFinAcknowledged", &this->tcpip_traps[1], tcpe_cb);
-    register_tcpip_trap(drakvuf, tcpip_profile_json,  "TcpDisconnectTcb", &this->tcpip_traps[2], tcpe_cb);
-    register_tcpip_trap(drakvuf, tcpip_profile_json, "TcpShutdownTcb", &this->tcpip_traps[3], tcpe_cb);
-    register_tcpip_trap(drakvuf, tcpip_profile_json, "TcpSetSockOptTcb", &this->tcpip_traps[4], tcpe_cb);
-    register_tcpip_trap(drakvuf, tcpip_profile_json, "UdpSetSockOptEndpoint", &this->tcpip_traps[5], udpb_cb);
-    register_tcpip_trap(drakvuf, tcpip_profile_json, "TcpCreateListenerWorkQueueRoutine", &this->tcpip_traps[6], tcpl_cb);
+    event_response_t(*tcpe_cb)( drakvuf_t drakvuf, drakvuf_trap_info_t* info ) = nullptr;
+    if (pm == VMI_PM_IA32E)
+    {
+        switch (winver)
+        {
+            case VMI_OS_WINDOWS_8:
+                // Tested on Windows 8.1 update 1 x64
+                tcpe_cb = tcpe_win81_x64_cb;
+                break;
+            case VMI_OS_WINDOWS_10:
+                switch (get_win_build_number(drakvuf))
```

Comment:
How about instead of a switch, we just default to the newer layout if the build is `>= 1734`?

Reply:
@tklengyel, done!